rotkehlchen/db/snapshots.py

```diff
@@ -202,3 +218,81 @@ def _export(
             return False, str(e)
 
         return True, ''
+
+    def import_snapshot(
+        self,
+        balances_snapshot_file: Path,
+        location_data_snapshot_file: Path,
+    ) -> Tuple[bool, str]:
+        main_currency = self.db.get_main_currency()
+        balances_list = self._csv_to_dict(balances_snapshot_file)
+        location_data_list = self._csv_to_dict(location_data_snapshot_file)
+        # check if the headers match the type stored in the db
+        has_invalid_headers = (
+            tuple(balances_list[0].keys()) != ('timestamp', 'category', 'asset_identifier', 'amount', f'{main_currency.symbol.lower()}_value') or  # noqa: E501
+            tuple(location_data_list[0].keys()) != ('timestamp', 'location', f'{main_currency.symbol.lower()}_value')  # noqa: E501
+        )
+        if has_invalid_headers:
+            return False, 'csv file has invalid headers'
+
+        # check if all timestamps are the same.
+        balances_timestamps = [entry['timestamp'] for entry in balances_list]
+        location_data_timestamps = [entry['timestamp'] for entry in location_data_list]
+        has_different_timestamps = (
+            balances_timestamps.count(balances_timestamps[0]) != len(balances_timestamps) or
+            location_data_timestamps.count(location_data_timestamps[0]) != len(location_data_timestamps) or  # noqa: E501
+            balances_timestamps[0] != location_data_timestamps[0]
+        )
+        if has_different_timestamps:
+            return False, 'csv file has different timestamps'
+
+        # check if the timestamp format == '%Y-%m-%d %H:%M:%S'
+        try:
+            timestamp = create_timestamp(balances_list[0]['timestamp'])
+        except ValueError:
+            return False, 'csv file contains invalid timestamp format'
+
+        return self._import_snapshot(
+            balances_list=balances_list,
+            location_data_list=location_data_list,
+            timestamp=timestamp,
+        )
+
+    def _import_snapshot(
+        self,
+        balances_list: List[Dict[str, str]],
+        location_data_list: List[Dict[str, str]],
+        timestamp: Timestamp,
+    ) -> Tuple[bool, str]:
+        processed_balances_list = []
+        processed_location_data_list = []
+        main_currency, main_currency_price = self.get_main_currency_price(timestamp)
+
+        for entry in balances_list:
+            processed_balances_list.append(
+                DBAssetBalance(
+                    category=BalanceType.deserialize(entry['category']),
+                    time=create_timestamp(entry['timestamp']),
```

Comment:
Didn't we say that there would be two different files or parts in the snapshot export? One for usage by the end user that would be human readable, and one that would be for import/export that would be machine readable and thus use the timestamp and not a datetime?